server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/sql/Db2SqlVisitorHelper.java

```diff
@@ -150,4 +155,58 @@ public static Position getAdjustedEndPosition(
     }
     return end;
   }
+
+  public static ExecSqlWheneverNode.WheneverConditionType getConditionType(Db2SqlParser.Dbs_wheneverContext ctx) {
+    ParserRuleContext ruleContext = ((ParserRuleContext) ctx);
+    ExecSqlWheneverNode.WheneverConditionType conditionType = ExecSqlWheneverNode.WheneverConditionType.NOT_FOUND;
+
+
+    if (ruleContext.getChildCount() >= 3) {
+      ParseTree pt = ruleContext.getChild(1);
+      String value = pt.getText().trim().toUpperCase();
+
+      if (Objects.equals(value, "SQLERROR")) {
+        conditionType = ExecSqlWheneverNode.WheneverConditionType.SQLERROR;
+      } else if (Objects.equals(value, "SQLWARNING")) {
+        conditionType = ExecSqlWheneverNode.WheneverConditionType.SQLWARNING;
+      }
+    }
+    return conditionType;
+  }
+
+  public static Pair<ExecSqlWheneverNode.WheneverType, String> getWheneverType(Db2SqlParser.Dbs_wheneverContext ctx) {
+    ParserRuleContext ruleContext = ((ParserRuleContext) ctx);
+    Pair<ExecSqlWheneverNode.WheneverType, String> result = Pair.of(ExecSqlWheneverNode.WheneverType.CONTINUE, null);
+
+    if (ruleContext.getChildCount() > 3) {
+
+      int index = 2;
+      ParseTree pt = ruleContext.getChild(index);
+      String value = pt.getText().trim().toUpperCase();
+      if (Objects.equals(value, "FOUND")) {
+        index = 3;
+        pt = ruleContext.getChild(index);
+        value = pt.getText().trim().toUpperCase();
+      }
+
+      if (Objects.equals(value, "DO")) {
+        result = Pair.of(ExecSqlWheneverNode.WheneverType.DO, ruleContext.getChild(index + 1).getText());
+      } else if (Objects.equals(value, "GO")) {
+        if (ruleContext.getChildCount() > index + 2) {
+          result = Pair.of(ExecSqlWheneverNode.WheneverType.GOTO, getHandler(ruleContext, index + 2));
+        }
+      } else if (Objects.equals(value, "GOTO")) {
+        result = Pair.of(ExecSqlWheneverNode.WheneverType.GOTO, getHandler(ruleContext, index + 1));
+      }
+    }
```

Comment:
Where does the `DO` come from?

Why not just access `dbs_host_name_container` context and `dbs_host_names ` in it?

Reply:
More code, but why not